src/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/Context.cs

```diff
@@ -1242,11 +1242,14 @@ private static ContextOptions GetDefaultOptionForStore(ContextType storeType)
             {
                 return DefaultContextOptions.MachineDefaultContextOption;
             }
-            else
+            else if (storeType == ContextType.Domain || storeType == ContextType.ApplicationDirectory)
             {
-                Debug.Assert(storeType == ContextType.Domain || storeType == ContextType.ApplicationDirectory);
                 return DefaultContextOptions.ADDefaultContextOption;
             }
+
+            // The actual value of this is unimportant, as an invalid contextType will cause an
+            // exception to be thrown in the constructor later.
+            return ContextOptions.Negotiate;
```

Comment:
I don't understand this change.  If the problem was that the assert was triggering, then isn't this a potential behavior change?  And if it's not a behavioral change and the value doesn't matter in that case, why introduce the extra conditional checks and branches?